test/runner.sh

```diff
@@ -13,9 +13,11 @@ TEST_OUTPUT_DIR=${TEST_OUTPUT_DIR:-${EXE_DIR}}
 TEST_SUPPORT_FILE=${CURRENT_DIR}/sql/utils/testsupport.sql
 
 # PGAPPNAME will be 'pg_regress/test' so we cut off the prefix
-# to get the name of the test
+# to get the name of the test. To ensure TEST_DBNAME is a valid
+# postgres identifier, replace hyphens with underscores (this is
+# particularly important for template tests)
```

Comment:
This is not correct, `-` is a valid character in an identifier otherwise all the template tests wouldnt work. psql can quote variables for you like so: `:"TEST_DBNAME"`